src/mz/src/config_file.rs

```diff
@@ -139,6 +147,38 @@ impl ConfigFile {
         Ok(())
     }
 
+    /// Adds an app-password to the configuration file or
+    /// to the keychain if the vault is enabled.
+    #[cfg(target_os = "macos")]
+    pub fn add_app_password(
+        &self,
+        new_profile: &mut toml_edit::Table,
+        name: &str,
+        profile: TomlProfile,
+    ) -> Result<(), Error> {
+        if self.vault() == "keychain" {
```

Comment:
Is it possible to make the return value of `fn vault()` an enum? Seems cleaner than matching strings